### PR TITLE
build docker image

### PR DESCRIPTION
**Pull request by orymate**
Build a docker image with all the runtime deps of banzai cli that can be used in a ci flow or a kubernetes task.

- [x] document usage ( improved token and URL handling: #305 )
- [x] automatic build

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,3 +44,25 @@ For interactive login, just run `banzai login`, and follow the instructions give
 ### Use
 
 See [command reference](https://banzaicloud.com/docs/pipeline/cli/reference/) in the [official documentation](https://banzaicloud.com/docs/pipeline/cli/).
+
+### Use Docker image
+
+Logging in
```
Comment (orymate) [inline README.md:50]: We might add a version where no config file is mounted, and the base url and the token come from an env var.

**Review by pbalogh-sa — APPROVED** [review]

File: .github/workflows/docker.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+name: Docker Build
+on:
+  push:
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+"
```
Comment (pregnor) [inline .github/workflows/docker.yaml:5]: **Question**: are we planning to support draft tags such as `0.6.0-dev.1` on this repository?
(It's ok if we don't, just want to be conscious about it.)

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,30 @@
+ARG GO_VERSION=1.15
+ARG FROM_IMAGE=alpine:latest
+
+FROM golang:${GO_VERSION}-alpine3.12 AS builder
+RUN apk add --update --no-cache bash ca-certificates make curl git mercurial tzdata
+
+# Install kubectl
+ARG KUBECTL_VERSION=v1.16.1
+RUN curl -L -s https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl -o /usr/local/bin/kubectl
+RUN curl -L -s https://amazon-eks.s3-us-west-2.amazonaws.com/1.14.6/2019-08-22/bin/linux/amd64/aws-iam-authenticator -o /usr/local/bin/aws-iam-authenticator
```
Comment (pregnor) [inline Dockerfile:10]: **Question**: shouldn't we put the AWS IAM authenticator version (`1.14.6`) into an argument as well?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+RUN chmod +x /usr/local/bin/*
+
+RUN mkdir -p /build
+WORKDIR /build
+
+COPY go.* /build/
+RUN go mod download
+
+ARG VERSION
```
Comment (pregnor) [inline Dockerfile:20]: **Question**: is there any reason this argument is not at the beginning of the image?
It was unusual for me to encounter an `ARG` in the middle of an image, but then again I'm not a Docker expert, so I'm just curious.

**Review by pregnor — APPROVED** [review]
Looks good, please manual test it thoroughly.

File: README.md (hunk 1 above)
Comment (pbalogh-sa) [reply README.md:50]: https://github.com/banzaicloud/banzai-cli/issues/305

File: .github/workflows/docker.yaml (hunk 2 above)
Comment (orymate) [reply .github/workflows/docker.yaml:5]: I don't think we should publish images for those.

File: Dockerfile (hunk 4 above)
Comment (orymate) [reply Dockerfile:20]: this is a copy paste from pipeline's dockerfile :)
i think it affects caching (you can easily rebuild the same source tree with a new tag)

**Review by bonifaido — APPROVED** [review]

File: Dockerfile (hunk 4 above)
Comment (pregnor) [reply Dockerfile:20]: Makes sense.

File: Dockerfile (hunk 3 above)
Comment (pbalogh-sa) [reply Dockerfile:10]: Unfortunately, the date is contained by the URL, so the proper date should be passed as well.

Comment (pregnor) [reply Dockerfile:10]: Indeed.

**Pull request merged**